gwcs/coordinate_frames.py

```diff
@@ -444,21 +445,23 @@ def __init__(self, axes_order=(0,), reference_frame=None, unit=None,
     @property
     def _world_axis_object_classes(self):
         return {'spectral': (
-            u.Quantity,
+            coord.SpectralCoord,
             (),
             {'unit': self.unit[0]})}
 
     @property
     def _world_axis_object_components(self):
         return [('spectral', 0, 'value')]
 
-    def coordinates(self, *args, equivalencies=[]):
-        if hasattr(args[0], 'unit'):
-            return args[0].to(self.unit[0], equivalencies=equivalencies)
-        if np.isscalar(args):
-            return args * self.unit[0]
+    def coordinates(self, *args):
+        # using SpectralCoord
+        if isinstance(args[0], coord.SpectralCoord):
+            return args[0].to(self.unit[0])
         else:
-            return args[0] * self.unit[0]
+            if hasattr(args[0], 'unit'):
+                return coord.SpectralCoord(*args).to(self.unit[0])
```

Comment:
This hasn't linked the frame's reference frame with the SpectralCoord object.

Reply:
yes, it's incomplete, this still needs to be done.